NebulaWorld/Combat/CombatManager.cs

```diff
@@ -0,0 +1,263 @@
+﻿#region
+
+using System;
+using System.Collections.Generic;
+using System.Net.Sockets;
+using NebulaAPI.DataStructures;
+using NebulaModel.DataStructures;
+using NebulaModel.Packets.Combat.Mecha;
+using NebulaModel.Packets.Players;
+using UnityEngine;
+#pragma warning disable IDE1006 // Naming Styles
+
+#endregion
+
+namespace NebulaWorld.Combat;
+
+public class CombatManager : IDisposable
+{
+    public readonly ToggleSwitch IsIncomingRequest = new();
+
+    public static int PlayerId { get; private set; }
+    public static bool SerializeOverwrite { get; set; }
+
+    public struct PlayerPosition
+    {
+        public ushort id;
+        public int planetId;
+        public int starId;
+        public Vector3 position;
+        public VectorLF3 uPosition;
+        public bool isAlive;
+        public Mecha mecha;
+        public Vector3 skillTargetL;
+        public VectorLF3 skillTargetULast;
+        public VectorLF3 skillTargetU;
+    }
+
+    public PlayerPosition[] Players; // include self
+    public HashSet<int> ActivedPlanets;
+    public HashSet<int> ActivedStars;
+    public HashSet<int> ActivedStarsMechaInSpace; // player in the system and not on a planet
+    public Dictionary<int, int> IndexByPlayerId;
+
+    private PlayerAction_Combat actionCombat;
+    private static CombatManager instance;
+
+    public CombatManager()
+    {
+        Players = new PlayerPosition[2];
+        ActivedPlanets = [];
+        ActivedStars = [];
+        ActivedStarsMechaInSpace = [];
+        IndexByPlayerId = [];
+        instance = this;
+    }
+
+    public void Dispose()
+    {
+        PlayerId = 1;
+        actionCombat = null;
+        Players = null;
+        ActivedPlanets = null;
+        ActivedStars = null;
+        ActivedStarsMechaInSpace = null;
+        IndexByPlayerId = null;
+        instance = null;
+        GC.SuppressFinalize(this);
+    }
+
+    public void GameTick()
+    {
+        if (!Multiplayer.Session.IsGameLoaded)
+        {
+            return;
+        }
```

Comment:
Minor nitpick, could be simplified to:
```suggestion
        if (!Multiplayer.Session.IsGameLoaded) return;
```